build/includes/terraform.mk

```diff
@@ -16,54 +16,59 @@
+gcloud-terraform-cluster: GCP_CLUSTER_NAME ?= test-cluster
 gcloud-terraform-cluster: $(ensure-build-image)
 gcloud-terraform-cluster:
 ifndef GCP_PROJECT
 	$(eval GCP_PROJECT=$(shell sh -c "gcloud config get-value project 2> /dev/null"))
 endif
-	$(DOCKER_RUN) bash -c 'export TF_VAR_agones_version=$(AGONES_VERSION) && \
-		cd $(mount_path)/install/terraform && terraform apply -auto-approve -var values_file="" \
-		-var chart="agones" \
-	 	-var "cluster={name=\"$(GCP_CLUSTER_NAME)\", machineType=\"$(GCP_CLUSTER_NODEPOOL_MACHINETYPE)\", \
-		 zone=\"$(GCP_CLUSTER_ZONE)\", project=\"$(GCP_PROJECT)\", \
-		 initialNodeCount=\"$(GCP_CLUSTER_NODEPOOL_INITIALNODECOUNT)\"}"'
-	$(MAKE) gcloud-auth-cluster
+	$(DOCKER_RUN) bash -c 'cd $(mount_path)/examples/terraform-submodules/gke-local && \
+		 terraform apply -auto-approve  -var agones_version="$(AGONES_VERSION)" \
+		-var name=$(GCP_CLUSTER_NAME) -var machine_type="$(GCP_CLUSTER_NODEPOOL_MACHINETYPE)" \
+		-var values_file="" \
+		-var zone="$(GCP_CLUSTER_ZONE)" -var project="$(GCP_PROJECT)" \
+		-var node_count=$(GCP_CLUSTER_NODEPOOL_INITIALNODECOUNT)'
+	GCP_CLUSTER_NAME=$(GCP_CLUSTER_NAME) $(MAKE) gcloud-auth-cluster
 
 # Creates a cluster and install current version of Agones controller
 # Set all necessary variables as `make install` does
+# Unifies previous `make gcloud-test-cluster` and `make install` targets
 gcloud-terraform-install: GCP_CLUSTER_NODEPOOL_INITIALNODECOUNT ?= 4
 gcloud-terraform-install: GCP_CLUSTER_NODEPOOL_MACHINETYPE ?= n1-standard-4
 gcloud-terraform-install: ALWAYS_PULL_SIDECAR := true
 gcloud-terraform-install: IMAGE_PULL_POLICY := "Always"
 gcloud-terraform-install: PING_SERVICE_TYPE := "LoadBalancer"
 gcloud-terraform-install: CRD_CLEANUP := true
+gcloud-terraform-install: GCP_CLUSTER_NAME ?= test-cluster
 gcloud-terraform-install:
 ifndef GCP_PROJECT
 	$(eval GCP_PROJECT=$(shell sh -c "gcloud config get-value project 2> /dev/null"))
 endif
 	$(DOCKER_RUN) bash -c ' \
-		cd $(mount_path)/install/terraform && terraform apply -auto-approve -var agones_version="$(VERSION)" -var image_registry="$(REGISTRY)" \
+	cd $(mount_path)/examples/terraform-submodules/gke-local && terraform apply -auto-approve -var agones_version="$(VERSION)" -var image_registry="$(REGISTRY)" \
 		-var pull_policy="$(IMAGE_PULL_POLICY)" \
 		-var always_pull_sidecar="$(ALWAYS_PULL_SIDECAR)" \
 		-var image_pull_secret="$(IMAGE_PULL_SECRET)" \
 		-var ping_service_type="$(PING_SERVICE_TYPE)" \
 		-var crd_cleanup="$(CRD_CLEANUP)" \
-		-var "cluster={name=\"$(GCP_CLUSTER_NAME)\", machineType=\"$(GCP_CLUSTER_NODEPOOL_MACHINETYPE)\", \
-		 zone=\"$(GCP_CLUSTER_ZONE)\", project=\"$(GCP_PROJECT)\", \
-		 initialNodeCount=\"$(GCP_CLUSTER_NODEPOOL_INITIALNODECOUNT)\"}"'
-	$(MAKE) gcloud-auth-cluster
+		-var chart="../../../install/helm/agones/" \
+		-var name=$(GCP_CLUSTER_NAME) -var machine_type="$(GCP_CLUSTER_NODEPOOL_MACHINETYPE)" \
+		-var values_file="../../../install/helm/agones/values.yaml" \
```

Comment:
Curious - why are we passing in the value file here? It's defaulted as part of the helm config.

Reply:
I also thought to remove, just to make it more obvious and to avoid extra digging to the configs.

Reply:
Agree with you, it is better to remove. Updated PR